packages/adapter-drizzle/src/sqlite/index.ts

```diff
@@ -0,0 +1,223 @@
+  },
+  (vt) => ({
+    compoundKey: primaryKey(vt.identifier, vt.token),
+  })
+)
+
+
+export const db = drizzle(sqlite)
+export const defaultSchema = { users, accounts, sessions, verificationTokens }
+
+export type DbClient = typeof db
+
+export type DefaultSchema = typeof defaultSchema
+interface CustomSchema extends DefaultSchema { }
+
+export function SQLiteDrizzleAdapter(
+  client: DbClient,
+  schema?: Partial<CustomSchema>
+): Adapter {
+  const { users, accounts, sessions, verificationTokens } = {
+    users: schema?.users ?? defaultSchema.users,
+    accounts: schema?.accounts ?? defaultSchema.accounts,
+    sessions: schema?.sessions ?? defaultSchema.sessions,
+    verificationTokens:
+      schema?.verificationTokens ?? defaultSchema.verificationTokens,
+  }
+
+  return {
+    createUser: (data) => {
+      return client
+        .insert(users)
+        .values({ ...data, id: crypto.randomUUID() })
+        .returning()
+        .get()
+    },
+    getUser: (data) => {
+      return client.select().from(users).where(eq(users.id, data)).get() ?? null
+    },
+    getUserByEmail: (data) => {
+      return (
+        client.select().from(users).where(eq(users.email, data)).get() ?? null
+      )
+    },
+    createSession: (data) => {
+      return client.insert(sessions).values(data).returning().get()
+    },
+    getSessionAndUser: (data) => {
+      return (
+        client
+          .select({
+            session: sessions,
+            user: users,
+          })
+          .from(sessions)
+          .where(eq(sessions.sessionToken, data))
+          .innerJoin(users, eq(users.id, sessions.userId))
+          .get() ?? null
+      )
+    },
+    updateUser: (data) => {
+      if (!data.id) {
+        throw new Error("No user id.")
+      }
+
+      return client
+        .update(users)
+        .set(data)
+        .where(eq(users.id, data.id))
+        .returning()
+        .get()
+    },
+    updateSession: (data) => {
+      return client
+        .update(sessions)
+        .set(data)
+        .where(eq(sessions.sessionToken, data.sessionToken))
+        .returning()
+        .get()
+    },
+    linkAccount: (rawAccount) => {
+      const updatedAccount = client
+        .insert(accounts)
+        .values(rawAccount)
+        .returning()
+        .get()
+
+      const account: AdapterAccount = {
+        ...updatedAccount,
+        type: updatedAccount.type,
+        access_token: updatedAccount.access_token ?? undefined,
+        token_type: updatedAccount.token_type ?? undefined,
+        id_token: updatedAccount.id_token ?? undefined,
+        refresh_token: updatedAccount.refresh_token ?? undefined,
+        scope: updatedAccount.scope ?? undefined,
+        expires_at: updatedAccount.expires_at ?? undefined,
+        session_state: updatedAccount.session_state ?? undefined,
+      }
+
+      return account
+    },
+    getUserByAccount: (account) => {
+      const dbAccount =
```

Comment:
```suggestion
      const results = await client
       .select().from(accounts)
       .leftJoin(users, eq(users.id, accounts.id))
       .where(
         and(eq(accounts.provider, account.provider), eq(accounts.providerAccountId, account.providerAccountId))
       ).get()
       
       return results.users ?? null;
      
```

Reply:
This is likely more performant. I don't see any indexes made in this PR. Is that something that should be done as well?

Reply:
https://github.com/nextauthjs/next-auth/pull/7165/files#r1253130773 index suggestions here. There should be indexes for each type (pg, MySQL, sqlite) IMO

Reply:
Thanks!